adapter/internal/api/apis_impl.go

```diff
@@ -259,19 +262,25 @@ func ApplyAPIProjectFromAPIM(
 	conf, _ := config.ReadConfigs()
 	currentEnv := conf.ControlPlane.EnvironmentLabels[0] // assumption - adapter has only one environment
 
-	if apiEnvs[apiProject.APIYaml.Data.ID][currentEnv].APIConfigs.SandboxEndpointChoreo != "" {
-		vhostToEnvsMap[conf.Adapter.SandboxVhost] = []string{currentEnv}
+	if apiEnvs[apiProject.APIYaml.Data.ID][currentEnv].APIConfigs.SandboxEndpointChoreo != "" &&
+		conf.ControlPlane.DynamicEnvironments.Enabled {
```

Comment:
Why does this is applied only to dynamic environments?

Reply:
When dynamic environments are enabled we are already getting this as a separate deployment. Hence we need not to process this SandboxEndpointChoreo property. Here is a sample.
```
"6433948c644c6c4067d4405f": {
      "pdp-prod": {
        "envId": "a6a726fd-ee60-4aa7-8405-f3779494e70d",
        "envName": "pdp-prod",
        "configs": {
          "productionEndpoint": "https://run.mocky.io/v3/fc43a3db-0e14-4fd5-9692-a6fc8e07e01a",
          "sandboxEndpointChoreo": "https://run.mocky.io/v3/496df25f-6f3e-490f-a40f-aff90fc03907"
        }
      },
      "pdp-dev": {
        "envId": "5696593c-305a-4e76-bc9c-d5cbc079f4ae",
        "envName": "pdp-dev",
        "configs": {
          "productionEndpoint": "https://run.mocky.io/v3/fc43a3db-0e14-4fd5-9692-a6fc8e07e01a",
          "sandboxEndpointChoreo": ""
        }
      },
      "pdp-prod-sandbox": {
        "envId": "5abb54bf-0451-4d93-9a84-8489020f6291",
        "envName": "pdp-prod-sandbox",
        "configs": {
          "productionEndpoint": "https://run.mocky.io/v3/496df25f-6f3e-490f-a40f-aff90fc03907"
        }
      }
    },
    ```

Reply:
My bad it should be otherway around. Thanks for pointing it out.